### ISSUE TITLE
Is bolthold practical for time series?

### ISSUE THREAD

**Issue by cbrake** [timshannon/bolthold#62]
Hi, I'm considering using bolthold for time series data. Ideally, with a time series database, you can filter data by a time range and quickly pull out the data you want with having to query over the entire bucket. If I was building on boltdb directly, I could do something like create separate buckets for each week of the year or something. Do you have any thoughts if it is practical to store time series data using bolthold?

The application for this is simple setups where we are storing data from a handful of sensors over the course of years on an embedded Linux system, where we can't justify something heavier like influxdb. I think there is also an application for some http://simpleiot.org installations where someone wants to set up an instance to log data from a handful of sensors without messing with a separate DB.

**Comment by timshannon** [comment]
Whether your in Bolt or BoltHold, if your Key value is `time.Time`, it should allow you to query ranges without having to read the entire bucket, as Bolt supports Seeking, and BoltHold utilizes that in [range queries](https://github.com/timshannon/bolthold/blob/master/index.go#L290).

**Comment by cbrake** [comment]
@timshannon thanks for the information -- that sounds like a good fit then for what I need.

**Comment by cbrake** [comment]
I ran into problems using a Gob encoded Time.time for key, as it does not seem like they were always sequential with time. A bolt Seek does not always take me to where I want to go in the data sequence. I did not have time to investigate the reason, but ended up dropping down into bolt for this task and used a time.UnixNano int64 as the key -- that seems to be working fairly well.

I do have a question -- does bolthold encode int64's using the default gob encoder, or does it do a simpler big-endian conversion to bytes. The later might be useful in cases like this.

**Comment by timshannon** [comment]
Bolthold encodes the data however you want.  Be default it's Gob encoding, but you can use JSON, or MSGPack or your own custom binary encoding.

**Comment by cbrake** [comment]
I ended up encoding the time in the key using:

```
tNano := t.UnixNano()
b := make([]byte, 8) 
binary.BigEndian.PutUint64(b, uint64(tNano))
```

This encodes the time in big endian format so will be sequential as time increases in the DB, then bolt naturally sorts the time for you and Seek() works.

I then just used gob encoding for the sample value, so ended up using different encoding for key and value.

This is very simple and allows bolt to be used as a low end time series database. My interest in TSDs are not system that can scale out to billions devices with 100,000 writes/sec, but rather easy to install systems that can be run on IoT edge devices (Embedded Linux), or simple IoT backend apps that can service several thousand devices.
